### PR TITLE
Add example for `duplicate-argument-name` message documentation

### PR DESCRIPTION
**Pull request by gunungpw**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

add code example for `duplicate-argument-name`

Ref #5953

### CODE REVIEW

File: doc/data/messages/d/duplicate-argument-name/details.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.. code:: python
+
+    >>> def get_fruits(apple, banana, apple):  # [duplicate-argument-name]
+    ...     pass
+    ...
+    File "<stdin>", line 1
```
Comment (DanielNoord) [inline doc/data/messages/d/duplicate-argument-name/details.rst:6]: Schermafbeelding 2022-04-01 om 09 06 33

This is kind of duplicating what we already state above. We note that these raise SyntaxErrors and give this code example. Would you be okay with excluding this?

The `pylint-errors` project combines some of the data we originally provided so things that we have in our `Description` section can add up in their `Details` section.

File: doc/data/messages/d/duplicate-argument-name/related.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+`Testcases <https://github.com/PyCQA/pylint/blob/master/tests/functional/d/duplicate_argument_name.py>`_
```
Comment (DanielNoord) [inline doc/data/messages/d/duplicate-argument-name/related.rst:1]: I understand why `pylint-error` has these in their examples, but I'm a bit scared that these links can easily change if we move tests around.
The Issue Tracker link also doesn't really capture all issues about this message if people use other names in their issues so I think it is not fool-proof.

**Review by DanielNoord — COMMENTED** [review]
@gunungpw Thanks for you first contribution to `pylint`!

I made some comments. I should probably have addressed this in the guide in the original issue. I will update it with these comments as well.

File: doc/data/messages/d/duplicate-argument-name/related.rst (hunk 2 above)
Comment (gunungpw) [reply doc/data/messages/d/duplicate-argument-name/related.rst:1]: Understandable 👍, I will remove it

File: doc/data/messages/d/duplicate-argument-name/details.rst (hunk 1 above)
Comment (gunungpw) [reply doc/data/messages/d/duplicate-argument-name/details.rst:6]: Okay, thanks for the advice. I will also remove it 😄

**Review by DanielNoord — APPROVED** [review]

File: pylint/checkers/base/basic_error_checker.py [hunk 3]
```diff
@@ -285,14 +285,17 @@ def visit_functiondef(self, node: nodes.FunctionDef) -> None:
 
         # provide detailed report about each repeated argument
         for argument_duplicates in arg_clusters.values():
```
Comment (DanielNoord) [inline pylint/checkers/base/basic_error_checker.py:287]: @Pierre-Sassoulas Can't we just merge this with for loop above? Add `arg.name` to a list. If `name` is already in that list raise a message on the current `arg` node being checked?

Comment (Pierre-Sassoulas) [reply pylint/checkers/base/basic_error_checker.py]: Yeah thank you, I was in the mindset of extending the existing code, but a refactor is a lot better.

**Pull request merged**